exchanges/stream/websocket.go

```diff
@@ -1241,3 +1257,38 @@ func signalReceived(ch chan struct{}) bool {
 		return false
 	}
 }
+
+// GetConnection returns a connection by connection signature (defined in wrapper setup) for request and response
+// handling in a multi connection context.
+func (w *Websocket) GetConnection(connSignature any) (Connection, error) {
```

Comment:
I disagree this will add in more complexity than it needs to. The metadata can be extracted from the connection itself after an immutable key is associated with that connection type, I have tried the comparable type but it adds more complexity with no added benefit (unless you have an actual diff please show me of it).

With regards to asset type only that's not a bad idea but I have Bybit PR which is using a string as key for a Trade connection and a specific private connection for incoming data. 

As for more options for people using fancier signatures they can define a struct with fields with immutable keys right now if they so choose.

Reply:
What I can do in the meantime is do a comparable check on the keys being set if not nil using reflection as an option.

Reply:
I agree that `any`thing is more complex than `any`.  I cannot see anywhere one can extract the asset from a connection. A connection itself doesn't even have a signature. The only way I can see that you can retrieve the asset is from `Websocket.GetConnection` and _hope_ that the signature is simply an asset and if you get a connection back. 

So far the requirements you have told me for a connection signature is a `string` and an `asset.Item` and struct containing those two does not sound complex. I don't see how utilising reflection is simpler

I'm still open to using `any`, but I think I'll need more convincing. What I think is needed is a way to get connection signatures. At present you can only return a connection based on if you already know the signature in advance. Then you can dynamically determine the connection you want. At least with a type you have clarity and structure for what is expected of a signature. That's where I think `any` becomes more problematic than a type, but it can still work, just with more complexity. Being able to dynamically retrieve a connection I think would be to the feature's benefit.

Reply:
> I agree that anything is more complex than any. I cannot see anywhere one can extract the asset from a connection. A connection itself doesn't even have a signature. The only way I can see that you can retrieve the asset is from Websocket.GetConnection and hope that the signature is simply an asset and if you get a connection back.

Why do you need that anyway? There is no way to get a list of connections, everything is set and handled in wrapper? I must be confused on what your requesting as I have said above `I have tried the comparable type but it adds more complexity with no added benefit (unless you have an actual diff please show me of it)` So please send me an example or a diff and what system will need that if any. 

> So far the requirements you have told me for a connection signature is a string and an asset.Item and struct containing those two does not sound complex. I don't see how utilising reflection is simpler. At least with a type you have clarity and structure for what is expected of a signature.

This is the only check I have added which forces comparable on the key which is wrapper defined and not used externally [here](https://github.com/thrasher-corp/gocryptotrader/pull/1598/commits/511c78b43522acd6fe8b0b077896070439f2fc0f) This is only for setting the comparable key. This does not check every time you call get connection. 

> I'm still open to using any and maybe I'm misunderstanding everything you're saying. What I think is needed is a way to get connection signatures (even if its returning []any). At present you can only return a connection based on if you already know the signature in advance. The feature for returning signatures you can then dynamically determine the connection you want. That's where I think any becomes more problematic than a type, but it can still work, just with more complexity for the caller. Being able to dynamically retrieve connections I think would be to the feature's benefit.

There is yet to be any external request path introduced in this PR so what your requesting can't even be tested properly, everything is handle via wrapper functions, the decision to use the connection is consolidated to the wrapper and not external. Can you open PR based off this and we can go down that direction when there is a use case? I don't see an issue its just I am reluctant to add things in the off chance its going to be used. Or maybe I am still not understanding the request.

Reply:
I think this is a result of exporting functions like `AuthenticateSpot(ctx context.Context, conn stream.Connection) error` which is designed and developed specifically for the handling inside this package and makes no mention of it. When I'm seeing exported functions that ask for a `conn` with the only way to get a connection is to call _this exported function_ `GetConnection`, where the only way to retrieve a connection is to already know the hard-coded connection signature that can be of any type. I hope from that its a bit clearer about why I've been saying what I've been saying

I think the best way of closing this is unexporting those two functions `AuthenticateSpot` and `WebsocketLogin`. You are passing `AuthenticateSpot` as a function already and it closes the funniness a bit and ensures that your stream package functions are doing the handling of it all.

Otherwise, sounds like we need to talk IRL to get a better understanding of things 🌈

Reply:
[unexported](https://github.com/thrasher-corp/gocryptotrader/pull/1598/commits/d3343a7dbde3a935286a248b1bf19c8ed6524b99)